parsers/c/c_lexer.mll

```diff
@@ -7,16 +7,55 @@ open Tokens
 exception Error of Errors.cparser_cause
 
 type flags = {
-  inside_cn : bool;
-  magic_comment_char : char;
-  at_magic_comments : bool;
+  inside_cn : bool;           (* We are lexing in a CN comment *)
+  magic_comment_char : char;  (* The character after a comment indicating to start a CN comment *)
+  at_magic_comments : bool;   (* Should we process CN comments (true) or treat them as normal comments (false) *)
 }
 
+(* WARNING: GLOBAL STATE
+We wouldn't need this if the Lexing.position type was parameterized on the
+type of locations or if there was some way to extend the lexer buffer with
+additional user state, but I can't seem to find a way to do either of these,
+so for the time being we use a global variable.
+
+Note that one needs to call `init` when starting to work a
+new (preprocessed) file.
+*)
+module LineMap = struct
+
+  (** Maps a line in the preprocessed file to a filename and a line number.
+  The following lines in the preprocessed file reside at the specified source
+  location, until the next entry in the map. *)
+  let mapping = ref (Cerb_position.LineMap.empty "")
+
+  (** Clear all line mapping entries *)
+  let init file = mapping := Cerb_position.LineMap.empty file
+
+  (** Add a new line mapping *)
+  let add line info = mapping := Cerb_position.LineMap.add line info !mapping
+
+  (** Get the current mapping *)
+  let get () = !mapping
+
+  (** Make a position from a Lexing.position, consulting the line map *)
+  let position x =
+    let p1 = Cerb_position.from_lexing x in
+    let line_map = !mapping in
+    let src_loc = Cerb_position.LineMap.lookup (Cerb_position.line p1) line_map in
+    Cerb_position.set_source src_loc p1
+
+end
```

Comment:
This is definitely possible, and only looks like few lines of code - would you like to try this in a follow-up PR?
https://schu.be/til-sedlex-and-menhir.html
https://gitlab.inria.fr/fpottier/menhir/blob/master/lib/Convert.mli#L69

Reply:
Ah, nice!  I'll certainly try to do this, but I think we should wait until the repository split.